pkg/plugins/plugins.go

```diff
@@ -214,7 +218,7 @@ func loadWithRetry(name string, retry bool) (*Plugin, error) {
 			}
 
 			timeOff := backoff(retries)
-			if abort(start, timeOff) {
+			if abort(start, timeOff, testTimeOut) {
```

Comment:
I'm wondering what's the point of the exponential backoff retry here. Is it used for restore scenarios, to give plugins enough time to initialize?

Reply:
Good question; as to the "exponential", I think that's a good pattern (although randomised _and_ exponential would be even better); we did have some cases where retry-loops could cause significant load on a server.

As to "why the retry at all", I had to look that up. It looks like it was added in https://github.com/moby/moby/commit/45fdce8a0d36d9b0825a6dfb6634a81f9af258af ;

- https://github.com/moby/moby/pull/15921

It looks indeed that it was added to prevent containers not being started / restored if the plugin was not yet up and running.

I see there even was a follow-up for networkingking plugins (related to https://github.com/moby/libnetwork/issues/813), but that was rejected; https://github.com/moby/moby/pull/19390